### PR TITLE
adds tsconfig support

### PR DESCRIPTION
**Pull request by khrome83**
This should solve this issue - 
#25 

I also added support for this issue because it was an adjacent problem.
#23 

This still does not replicate the whole CLI options. But I think I covered any that would be meaningful.

### CODE REVIEW

**Comment by khrome83** [comment]
@lucacasonato let me know if there is anything else I need to do. I also updated the `README.md` to reflect the additions.

File: src/index.ts [hunk 1]
```diff
@@ -89,7 +112,38 @@ async function buildDenoLambda(
...
+        ...(allowWrite && typeof allowWrite === 'string'
+          ? [`--allow-write=${allowWrite}`]
+          : []),
+        ...(allowNet && typeof allowNet === 'boolean' ? ['--allow-net'] : []),
+        ...(allowNet && typeof allowNet === 'string'
+          ? [`--allow-net=${allowNet}`]
+          : []),
+
+        // Strings
+        ...(tsConfig ? ['-c', tsConfig] : []),
```
Comment (Nexxkinn) [inline src/index.ts]: --config flag is only available from 1.0.0 upwards.
Consider adding a version check to prevent unknown flag error.

Comment (khrome83) [reply src/index.ts]: good catch, ty

Comment (khrome83) [reply src/index.ts]: Fixed

**Review by lucacasonato — COMMENTED** [review]
@khrome83 Thanks for the PR. You added all of these permission flags to the bundling command though, not to the actual run command. That is not necessary anyway because all functions are already run with `--allow-all`.

~~The config flag would in this case only be used during bundling also. If you want to support the config flag during runtime, this needs to be supported upstream in hayd/deno-lambda.~~ It is already supported by hayd/deno-lambda

**Comment by khrome83** [comment]
Ok @lucacasonato - that makes sense... 

I am still seeing a permissions issue when running `vercel dev`. I think there is a bug somewhere then. See #25

**Comment by khrome83** [comment]
I will update the PR to add support for the config then, and remove the permissions changes.

**Comment by khrome83** [comment]
Bump

**Comment by lucacasonato** [comment]
@khrome83 I’ll review shortly. Very busy at the moment.

**Comment by khrome83** [comment]
Understood. Take your time.

[Zane Milakovic - Chat @ Spike](https://spikenow.com/r/a/?ref=spike-organic-signature&_ts=itvjf)	[itvjf]

On May 24, 2020 at 16:27 GMT, Luca Casonato <notifications@github.com> wrote:

[@khrome83](https://github.com/khrome83) I’ll review shortly. Very busy at the moment.

File: src/index.ts [hunk 2]
```diff
@@ -89,7 +92,21 @@ async function buildDenoLambda(
...
+        entrypointPath,
+        binPath,
+        // Boolean
+        ...(unstable ? ['--unstable'] : []),
+        // Strings
+        // Deno Version 0.42.0 will parse as 0.42, 1.0.0 will parse as 1
+        // Quick fix for deno version. In the future a better "version check"
+        // could be built. But for right now this fixes the issue.
+        ...(tsConfig &&
+        (DENO_VERSION === 'latest' || parseFloat(DENO_VERSION) >= 1)
```
Comment (lucacasonato) [inline src/index.ts:106]: Nice hack :smile:

**Review by lucacasonato — APPROVED** [review]
Looks great, could you remove the package-lock.json though? This project uses `yarn` so there is a `yarn.lock` :-)

File: src/index.ts (hunk 2 above)
Comment (Nexxkinn) [reply src/index.ts:106]: Should we parse deno version for future case? Since it's using semver format, we could easily parse it to `Major.minor.patch`

Comment (lucacasonato) [reply src/index.ts:106]: No, it's good enough for now. We can fix it when it breaks :-)

**Pull request merged**